CRM/Upgrade/Incremental/MessageTemplates.php

```diff
@@ -0,0 +1,140 @@
+    return $this->upgradeVersion;
+  }
+
+  /**
+   * @param string $upgradeVersion
+   */
+  public function setUpgradeVersion($upgradeVersion) {
+    $this->upgradeVersion = $upgradeVersion;
+  }
+
+  /**
+   * CRM_Upgrade_Incremental_MessageTemplates constructor.
+   *
+   * @param string $upgradeVersion
+   */
+  public function __construct($upgradeVersion) {
+    $this->setUpgradeVersion($upgradeVersion);
+  }
+
+  /**
+   * Get any templates that have been updated.
+   *
+   * @return array
+   */
+  protected function getTemplateUpdates() {
+    return [
+      [
+        'version' => '5.4.alpha1',
+        'template' => 'membership_online_receipt',
+        'upgrade_descriptor' => ts('Use email greeting at top where available'),
+        'type' => 'text',
+      ],
+    ];
+  }
+
+  /**
+   * Get any required template updates.
+   *
+   * @return array
+   */
+  public function getTemplatesToUpdate() {
+    $templates = $this->getTemplateUpdates();
+    $return = [];
+    foreach ($templates as $template) {
+      if ($template['version'] === $this->getUpgradeVersion()) {
+        $return[$template['template'] . '_' . $template['type']] = $template;
+      }
+    }
+    return $return;
+  }
+
+  /**
+   * Get the upgrade messages.
+   */
+  public function getUpgradeMessages() {
+    $updates = $this->getTemplatesToUpdate();
+    $messages = [];
+    foreach ($updates as $key => $value) {
+      $messages[$key] = $value['upgrade_descriptor'];
+    }
+    return $messages;
+  }
+
+  /**
+   * Update message templates.
+   */
+  public function updateTemplates() {
+    $templates = $this->getTemplatesToUpdate();
+    foreach ($templates as $template) {
+      $workFlowID = CRM_Core_DAO::singleValueQuery("SELECT MAX(id) as id FROM civicrm_option_value WHERE name = %1", [
+        1 => [$template['template'], 'String'],
+      ]);
+      $content = file_get_contents(\Civi::paths()->getPath('[civicrm.root]/xml/templates/message_templates/' . $template['template'] . '_' . $template['type'] . '.tpl'));
+      $templatesToUpdate = [];
+      $templatesToUpdate[] = CRM_Core_DAO::singleValueQuery("SELECT id FROM civicrm_msg_template WHERE workflow_id = $workFlowID AND is_reserved = 1");
+      $defaultTemplateID = CRM_Core_DAO::singleValueQuery("
+        SELECT default_template.id FROM civicrm_msg_template reserved
+        LEFT JOIN civicrm_msg_template default_template
+          ON reserved.workflow_id = default_template.workflow_id
+        WHERE reserved.workflow_id = $workFlowID
+        AND reserved.is_reserved = 1 AND default_template.is_default = 1 AND reserved.id <> default_template.id
+      ");
+      if ($defaultTemplateID) {
+        $templatesToUpdate[] = $defaultTemplateID;
+      }
+
+      CRM_Core_DAO::executeQuery("
+        UPDATE civicrm_msg_template SET msg_{$template['type']} = %1 WHERE id IN (" . implode(',', $templatesToUpdate) . ")", [
+          1 => [$content, 'String']
+          ]
```

Comment:
Is this indented correctly?

Reply:
hmm - I can't imagine jenkins will fail to weigh in in a minute :-)